pep-0699.rst

```diff
@@ -64,10 +64,12 @@ Certain extensions use ``ma_version_tag`` for fast dictionary or globals
 lookups. For example,
 `Cython uses the field for fast dynamic module variable lookups <https://github.com/cython/cython/blob/169876872f3cb6198971a1db07e5b8a9d12b3dac/Cython/Utility/ObjectHandling.c#L1556>`_.
 
-Since this field is private and makes no guarantees on backwards compatibility,
-this PEP proposes to repeal :pep:`509` immediately within the next CPython
-version. A function for fast cached dictionary lookups may be provided as an
-alternative feature to replace directly accessing ``ma_version_tag``.
+This PEP proposes to emit a compiler warning when accessing ``ma_version_tag``.
+After two consecutive version releases with warnings, ``ma_version_tag``
+may be changed or removed, in line with :pep:`387`.
+A function for fast cached dictionary
```

Comment:
Can you drop this section? Saying that we may do it is saying nothing, and I don't see any value in doing this.

We already have dict watchers, so this is unnecessary for global dicts.
For object dicts, we could offer a "with cache" version of `PyObject_GetAttr()` but by then, "write it in Python" will be better advice for code with lots of dynamic attribute lookup.